wbooks/src/app/screens/Library/components/BookData/index.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { View, Text, TouchableOpacity, Image, ListRenderItem } from 'react-native';
+import bcInicio from '@assets/General/bc_inicio.png';
+
+import styles from './styles';
+
+interface Props {
+  id: number;
+  author: string;
+  title: string;
+  genre: string;
+  publisher: string;
+  year: string;
+  imageUrl: string | null;
+}
+
+const BookData: ListRenderItem<Props> = ({ item }) => {
```

Comment:
You are typing this as "ListRenderItem" so, this components can be used only inside of a renderItem. That is fine but you defined this "BookData" as a componente of Library (for all screen inside of Library) and perhaps only is a component of BookList.

Reply:
I restructured the branch, now I manage only one screen "Library" (component that contains the FlatList) and within this screen, there is a component name "BookData" (the component that renders the data). In the logic that I propose, there are no more screens inside "Library".